yarn-project/archiver/src/archiver/archiver.ts

```diff
@@ -182,8 +182,12 @@ export class Archiver extends EventEmitter implements ArchiveSource, Traceable {
       throw new Error('Archiver is already running');
     }
 
+    await this.blobSinkClient.testSources();
+
     if (blockUntilSynced) {
-      await this.syncSafe(blockUntilSynced);
+      while (!(await this.syncSafe(true))) {
+        this.log.info(`Retrying initial archiver sync in ${this.config.pollingIntervalMs}ms`);
```

Comment:
Missing `sleep(this.config.pollingIntervalMs)` or does it happen elsewhere?

Reply:
Hah, good catch!